---
title: "A bio-functional polymer that prevents retinal scarring through modulation of NRF2 signalling pathway"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r imports}
library(GEOquery)
library(DESeq2)
library(dplyr)
library(ComplexHeatmap)

library(tidyverse)
library(vsn)
library(data.table)

library(ggplot2)
library(RColorBrewer)
library(pheatmap)

# Set the working directory
#setwd("/Users/seymour/Desktop/Bioinformatics/Advanced_Computational_Biology_BIOL0050/Week3/Gene_Exp_analysis_3/")
```




```{r}
gse <- getGEO(filename = "GSE176513_series_matrix.txt.gz", GSEMatrix = TRUE, getGPL = FALSE)
```


```{r}
# Create a data frame with samples and conditions
sample_info <- pData(gse)
condition_df <- data.frame(Sample_ID = sample_info$description, Condition_name = sample_info$title) %>%
  mutate(Sample_ID = gsub("gene_count_matrix.csv --", "", Sample_ID)) %>%
  mutate(Condition = substr(Sample_ID, 1, 1)) %>%
  mutate(Duration = substr(Sample_ID, 4, nchar(Sample_ID))) %>%
  mutate(cond_duration = paste(Condition, Duration, sep = "_"))  %>%
  mutate(Condition_name = substr(Condition_name, 1, nchar(Condition_name) - 4)) %>%
  mutate(Condition_name_short = gsub(" \\(8h\\)| \\(24h\\)", "", Condition_name))

```


```{r}
#Separate the 8h and 24h samples
condition_df_8h <- condition_df[condition_df$Duration == "8h", ]
condition_df_24h <- condition_df[condition_df$Duration == "24h", ]
```


```{r}
#Read in the count data
count_data <- read.csv(gzfile("GSE176513_gene_count_matrix.csv.gz"))
```


```{r}
col_order<- condition_df$Sample_ID
rownames(count_data) <- count_data$EnsemblID
aligned_count_data <- count_data[, col_order]
 
```


```{r}
all_dds <- DESeqDataSetFromMatrix(countData = aligned_count_data, colData = condition_df, design = ~ Condition)
all_dds <- all_dds[ rowSums(counts(all_dds)) > 1, ]
all_dds <- estimateSizeFactors(all_dds) 
```


```{r}
vsd <- vst(all_dds, blind = FALSE)
sampleDists_vsd <- dist(t(assay(vsd)))
sampleDistMatrix_vsd <- as.matrix( sampleDists_vsd )
rownames(sampleDistMatrix_vsd) <- paste(vsd$Condition_name)
colnames(sampleDistMatrix_vsd) <- paste(vsd$Condition_name)
```

### Distance matix of gene expression profiles from RNA sequencing of all four groups (Media only, 1 wt% poly(CEP), TNT and 1 wt% poly(CEP) + TNT at two time points (8 and 24 h), each done in triplicates. 24 h), each done in triplicates. The intensity of each box corresponds to the distance value shown in the legend.

```{r}

colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
heatmap <- pheatmap(sampleDistMatrix_vsd,
                    clustering_distance_rows = sampleDists_vsd,
                    clustering_distance_cols = sampleDists_vsd,
                    col = colors)
heatmap
```


```{r}
heatmap <- pheatmap(sampleDistMatrix_vsd,
                    clustering_distance_rows = sampleDists_vsd,
                    clustering_distance_cols = sampleDists_vsd,
                    col = colors,cellwidth=8,
                    cellheight=8,treeheight_row = 8, treeheight_col = 8, fontsize_row = 7,fontsize_col = 7)

```


```{r}
pca_data_condition<-plotPCA(vsd, intgroup="Condition_name_short")
pca_data_condition
```


```{r}
pca_data_duration<-plotPCA(vsd, intgroup="Duration")
pca_data_duration
```


```{r}
library(ggfortify)
library(patchwork)

pca_data_duration <- plotPCA(vsd, intgroup = "Duration")
pca_data_condition <- plotPCA(vsd, intgroup = "Condition_name_short")
match(pca_data_duration$data$name,pca_data_condition$data$name)
pca_data_condition$data$Duration<-pca_data_duration$data$Duration
pcaData<-pca_data_condition$data
ggplot(pcaData, aes(x= PC1, y = PC2))+geom_point(size= 3, aes(shape=Duration, fill=Condition_name_short))+scale_shape_manual(values=c(21, 25))+ scale_fill_manual(values = c("#E69F00", "#56B4E9","Red","Green"),guide = guide_legend(override.aes = list(shape = 22)))+scale_color_discrete()+labs(x = paste0("PC", pca_data_condition$x[, 1]), y = paste0("PC", pca_data_condition$x[, 2])) +ggtitle("PCA Plot - Condition") +theme_minimal()


plot_duration <- autoplot(pca_data_duration, data = vsd$Duration, shape  = "Duration") +
  labs(x = paste0("PC", pca_data_duration$x[, 1]), y = paste0("PC", pca_data_duration$x[, 2])) +
  ggtitle("PCA Plot - Duration") +
  theme_minimal()

# PCA plot with intgroup="Condition_name_short"
pca_data_condition <- plotPCA(vsd, intgroup = "Condition_name_short")
plot_condition <- autoplot(pca_data_condition, data = vsd$Condition_name_short, colour = "Condition_name_short", shape="Duration") +
  labs(x = paste0("PC", pca_data_condition$x[, 1]), y = paste0("PC", pca_data_condition$x[, 2])) +
  ggtitle("PCA Plot - Condition") +
  theme_minimal()

# Combine the two plots
combined_plot <- plot_duration + plot_condition

# Display the combined plot
print(combined_plot)

```



```{r}
#Create a function to investigate number of significant genes that are differentially expressed between conditions
create_dds <- function(condition_xh) {
  col_order <- condition_xh$Sample_ID
  aligned_count_data <- count_data[, col_order]
  dds <- DESeqDataSetFromMatrix(countData = aligned_count_data, colData = condition_xh, design = ~ Condition)
  dds <- dds[ rowSums(counts(dds)) > 1, ]
  dds <- estimateSizeFactors(dds)
  
  dds <- DESeq(dds)
  
  res<-results(dds, contrast=c("Condition","A","B"))
  sigs<-na.omit(res)
  sigs<-sigs[sigs$padj<0.05,]
  cat("Number of genes sig in poly(CEP) alone compared to media only:", nrow(sigs), "\n")
  #print(nrow(sigs))
  
  res1<-results(dds, contrast=c("Condition","C","D"))
  sigs1<-na.omit(res1)
  sigs1<-sigs1[sigs1$padj<0.05,]
  cat("Number of genes sig in poly(CEP)+TNT compared to TNT:", nrow(sigs1), "\n")
  
  return(dds)
}
```


```{r}
dds_8h <- create_dds(condition_df_8h)
```

```{r}
dds_24h <- create_dds(condition_df_24h)
```
### Question 3: I'm trying to replicate the Hierarchical clustering heatmaps of differentially expressed genes. Fig taken from supplementary fig. Code that follows is my attempt at it. Not sure if Heatmap is the best way to use as I can't find arguments to select the type of clustering method directly, ie using centroid, average or furthest.. 


![Fig](/Users/seymour/Desktop/Bioinformatics/Advanced_Computational_Biology_BIOL0050/Week3/Gene_Exp_analysis_3/HC_heatmap.png)
### Also is it possible to reduce the number of dendogram branches (row-wise) as mine seem to be too many compared to the published fig above. 

### Answer 3: The supplementary says "Hierarchical clustering heatmaps of differentially expressed genes (Wald test, Benjamini-Hochberg (BH) Padj < 0.05) identified by comparing all four groups at 8 and 24 h". You are trying to create a heatmap for one of the comparisons (D vs A), so the number of genes is obviously different from the genes plotted in 7c. Please do this for all comparisons and try to plot a heatmap for common genes. pheatmap is what has been used in the paper, I have given a glimpse of how to proceed. 

I think because the genes you are trying to plot are different, the clustering looks different, you can try k means clustering to control the number of clusters (again makes more sense to let hclust decide).

```{r}
res_8h1<-results(dds_8h,contrast=c("Condition","B","A"))

sigs_8h1 <- na.omit(res_8h1)
sigs_8h1 <- sigs_8h1[sigs_8h1$padj < 0.05,]
df_sigs_8h1 <- as.data.frame(sigs_8h1)
mat_8h1<-counts(dds_8h,normalized=T)[rownames(df_sigs_8h1),]
```

```{r}
counts_mat<-counts(dds_8h,normalized=T)
rownames(counts_mat) <- rownames(counts(res_8h1))
rownames(res_8h1)
```
```{r}
row_names <- rownames(results(res_8h1))
print(row_names)
```


```{r}
counts(dds_8h,normalized=T)
```



```{r}

#res_8h <- results(dds_8h)

res_8h1<-results(dds_8h,contrast=c("Condition","B","A"))
res_8h2<-results(dds_8h,contrast=c("Condition","C","A"))
res_8h3<-results(dds_8h,contrast=c("Condition","D","A"))



sigs_8h1 <- na.omit(res_8h1)
sigs_8h1 <- sigs_8h1[sigs_8h1$padj < 0.05,]
df_sigs_8h1 <- as.data.frame(sigs_8h1)
mat_8h1<-counts(dds_8h,normalized=T)
#[rownames(df_sigs_8h1),]
z_mat_8h1<-t(apply(mat_8h1,1,scale))
colnames(z_mat_8h1)<-condition_df_8h$Condition_name_short

Heatmap(z_mat_8h1,cluster_rows=T,column_labels=colnames(z_mat_8h1),show_row_names = FALSE)
pheatmap(z_mat_8h1,clustering_distance_rows = "euclidean",clustering_distance_cols = "euclidean",clustering_method = "complete",treeheight_row = 5,show_rownames = F)
```
```{r}
print(dim(df_sigs_8h1))
mat_8h1<-counts(dds_8h,normalized=T)[rownames(df_sigs_8h1),]
```
```{r}
count_matrix<- counts(dds_8h, normalized = TRUE)
count_matrix
```


```{r}
new_row_names <- gsub(",", "", rownames(count_matrix))
print(new_row_names)
rownames(count_matrix) <- new_row_names
```


```{r}
rownames(df_sigs_8h1)
```



```{r}
Heatmap(z_mat_8h,cluster_rows=T,column_labels=colnames(z_mat_8h),show_row_names = FALSE,method = "centroid")
```

```{r}
```


```{r}

```


```{r}

```


```{r}
```


```{r}
```


```{r}
```


```{r}
volcanoP <- ggplot(results_order, ggplot2::aes(log2FoldChange, -log10(padj))) +
  geom_point(ggplot2::aes(col = sig)) +
  scale_color_manual(values = c("red", "black")) +
  ggtitle("Significant genes Untr vs. dexamethasone")

volcanoP
```

