# Computational-biology

In [Project 1](https://rpubs.com/Seymour22/biofunctional_polymer_prevents_retinal_scarring) I use DeSeq2 to extract differential expressed genes in epithelial cells treated under four conditions: Media only control, Polymer alone, TNF-α and TGF-β1 (TNT) alone, TNT + Polymer. The purpose of this pipeline is to replicate the results published by [Parikh et al., 2022](https://pubmed.ncbi.nlm.nih.gov/35589753/) which showed that upregulation of genes that are important for the NRF2 signalling pathway is associated with the presence of the polymer and not necessarily TNT.


In [Project 2](https://rpubs.com/Seymour22/GWAS_analysis) I use Plink to perform a genome-wide association study to pinpoint the Single Nucleotide Polymorphisms (SNPs) that contribute to hippocampal atrophy in individuals diagnosed with Alzheimer's disease. The investigation utilizes genetic data extracted from the Alzheimer's Disease Neuroimaging Initiative (ADNI) dataset.

In [Project 3](https://rpubs.com/Seymour22/PRS_analysis) I perform Polygenic Risk Score (PRS) analysis using PRSice-2, where the objective is to explore the genetic architecture underlying complex traits of Alzheimer’s disease. PRS integrates information from multiple genetic variants (typically SNPs) which enables us to quantify an individual’s genetic predisposition for a specific trait or disease. 
